libs/shared-atomic-design-components/src/lib/molecules/Paginate.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Pagination,
+  PaginationContent,
+  PaginationEllipsis,
+  PaginationItem,
+  PaginationLink,
+  PaginationNext,
+  PaginationPrevious,
+} from '@talent-connect/shared-shadcn-ui-components'
+import './Paginate.scss'
+
+interface PaginateProps {
+  totalPagesNumber: number
+  currentPageNumber: number
+  setCurrentPageNumber: (page: number) => void
+}
+
+type ActivePagesType = number | 'ellipsis'
+
+const MAX_VISIBLE_PAGES = 5
+
+const Paginate = ({
+  totalPagesNumber,
+  currentPageNumber,
+  setCurrentPageNumber,
+}: PaginateProps) => {
+  if (!totalPagesNumber) return null
+
+  const scrollToListTop = () => {
+    window.scrollTo({ top: 350, behavior: 'smooth' })
+  }
+
+  const handlePreviousPage = () => {
+    if (currentPageNumber > 1) {
+      setCurrentPageNumber(currentPageNumber - 1)
+    }
+    scrollToListTop()
```

Comment:
Hmmm, TBH I don't enjoy the UX of being automatically sent to the top. And that only works with the imagined "happy path" where a company linearly wants to look through profiles page by page.

Imagine a scenario where someone wants to find their way back to page 5. They'd probably like to quickly click the right caret four times. That won't work with current functionality.

Reply:
What are your thoughts on this?

Reply:
@ericbolikowski, if there is no auto-scrolling to the top of the list, then the user always stays on the bottom (where the pagination buttons are) and has to scroll to the top on each page themselves. 

What am I not getting? :)

Reply:
Yep - it certainly is a trade-off. IMHO we should remove it. But I'm happy to defer to well-reasoned points like this one, and especially UX patterns. Did we do any research on this? Shouldn't be more than a few prompts away

Reply:
It seems you and I fall into two different user groups :)

FYI, here are the research results:

**Pros of the scroll behavior:**
The recommendation to scroll to the top when paginating is grounded in well-established UX principles, user expectations, and successful patterns observed in widely used websites and applications. Here's a breakdown of the sources and reasoning behind the recommendation to scroll to the top of the items list when a user clicks on a pagination button:

**1. User Experience Research**
Attention Focus: Research in UX shows that users generally expect to see new content at the top of the screen when they navigate to a new page. If the content doesn't automatically align with their field of vision, it can cause confusion and frustration, leading to a poor user experience.
Cognitive Load Reduction: By automatically scrolling to the top, you reduce the cognitive load on the user. They don't have to manually scroll to find where the new content starts, which makes the experience smoother and more intuitive.

**2. Consistency and Predictability**
Web Standards and Consistency: Many popular websites and applications, such as e-commerce sites and content platforms, implement automatic scrolling to the top when paginating. This consistency in behavior sets user expectations. Users come to expect that when they click to see a new page, they will be taken to the start of that page's content.
Usability Heuristics: Jakob Nielsen's heuristics emphasize consistency and standards as key factors in usability. Aligning with established patterns helps users feel comfortable and confident when interacting with a website.

**3. Case Studies and Real-World Examples**
E-commerce and Content Platforms: Websites like Amazon, eBay, and many news sites automatically scroll to the top of the content list when users click on a new page in a paginated list. These companies invest heavily in UX research and testing, indicating that this behavior aligns with user preferences.
User Testing and Feedback: In various usability tests, users have reported that failing to scroll to the top when paginating can lead to disorientation, especially if the pagination controls are near the bottom of the screen. This feedback has driven many UX designers to adopt auto-scrolling as a best practice.

**4. Accessibility Considerations**
Screen Reader and Keyboard Navigation: For users who rely on screen readers or keyboard navigation, automatically scrolling to the top of the new content ensures they can immediately start consuming the new page's content. This behavior is necessary for them to understand the context and find the beginning of the content.

**5. Smooth Transitioning**
Perceived Performance and Fluidity: Smooth scrolling enhances the perceived performance of an application by making transitions between content pages feel more fluid and responsive. This is especially important in modern web design, where users expect a certain level of polish in interactions.

**Summary**
This behavior is intended to provide a seamless, predictable, and accessible experience, reducing user frustration and improving overall satisfaction.

**Cons of the scroll behavior:**
There are scenarios where **not** scrolling to the top of the items list after a user clicks on a pagination button might be preferable, depending on the context and user needs. Here are some reasons why you might choose **not** to implement automatic scrolling:

**1. Context Preservation**
Maintaining User Flow: If the user is deeply engaged with content at the bottom of a list (e.g., reading detailed information or interacting with other UI elements), automatically scrolling to the top can be jarring and disruptive, breaking their flow.
Sequential Browsing: In situations where users are likely to review items in a sequence and don't need to start at the top of each new page, keeping the current scroll position might make it easier for them to continue from where they left off.

**2. Pagination Near the Top**
When Pagination Controls Are at the Top: If the pagination controls are placed at the top of the content list (or near the top in a fixed position), the user's current view might already be aligned with the start of the new content. In this case, automatic scrolling is unnecessary and causes confusion if the scroll position changes unexpectedly.

**3. Infinite Scroll Hybrid**
Combining Pagination with Infinite Scroll: Some designs use a hybrid approach where additional content is loaded dynamically as the user scrolls. In such cases, users might expect the new content to load in place without the page resetting to the top, allowing them to continue their browsing experience seamlessly.

**4. Partial Page Reloads**
Contextual Pagination: If pagination only reloads part of the page or the content within a specific section (like a table or grid within a dashboard), keeping the current scroll position might make more sense, mainly if the rest of the page contains important context or controls that the user is actively interacting with.

**5. User Control and Preferences**
Letting Users Decide: Some advanced interfaces allow users to control their navigation experience, including whether or not they want the page to scroll to the top after paginating. This could be implemented as a user preference setting, catering to different user behaviors.

**6. Avoiding Unintended User Frustration**
User Expectations in Specific Contexts: In some contexts, users might not expect the page to scroll automatically. For example, if users filter data or make detailed comparisons across pages, keeping the current scroll position could help them maintain context and avoid unnecessary scrolling.

**7. Low-Content Pages**
Short Content Lists: If the content list is concise (e.g., a list with only a few items per page), scrolling to the top might not be necessary because users can see all items without significant scrolling. In this case, keeping the position might be less disruptive.

**Summary**
Not scrolling to the top after paginating can be beneficial in contexts where preserving the user's current focus or position is more critical than resetting their view. This approach is often about maintaining continuity, avoiding disruption, and catering to specific user needs or expectations. The decision ultimately depends on the nature of the content, the structure of the UI, and the particular user tasks being supported.

What are your thoughts?

Reply:
@helloanil, @astkhikatredi, feel free to chime into our discussion above if you have an opinion on that.

Reply:
guys, I have a question - is it too much if we add also pagination buttons to the top of the page, what do you think?
Regarding auto-scrolling to the top of the list - I would personally like it

Reply:
I can't argue with user research - and the wall of text 😆 I'll happily defer to you peeps

Reply:
Well, you asked about research 🙂 
I provided you one, and now you don't even want to read it 🥺